overlord/snapstate/handlers_components.go

```diff
@@ -561,12 +565,6 @@ func (m *SnapManager) doUnlinkComponent(t *state.Task, _ *tomb.Tomb) (err error)
 		return err
 	}
 
-	// TODO:COMPS: test taking this branch when unlinking components during a
-	// refresh where we lose components
-	if err := saveCurrentKernelModuleComponents(t, snapSup, snapSt); err != nil {
```

Comment:
This was in the wrong place. We now do this in 3 places:
```
link-component
unlink-current-component
link-snap
```
Previously we had it in `unlink-component`, rather than `unlink-current-component`. `unlink-component` isn't run during a snap refresh, that is only run when we are explicitly removing a snap (with its components)

Reply:
In that case we use "unlink-current-component", since we don't support removing components for anything but the current snap revision.

Reply:
See here: https://github.com/canonical/snapd/blob/master/overlord/snapstate/component.go?plain=1#L560